src/test/groovy/edu/ie3/simona/model/participant/StorageModelTest.groovy

```diff
@@ -0,0 +1,374 @@
+/*
+ * © 2022. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.simona.model.participant
+
+import edu.ie3.datamodel.models.OperationTime
+import edu.ie3.datamodel.models.input.NodeInput
+import edu.ie3.datamodel.models.input.OperatorInput
+import edu.ie3.datamodel.models.input.system.StorageInput
+import edu.ie3.datamodel.models.input.system.characteristic.CosPhiFixed
+import edu.ie3.datamodel.models.input.system.type.StorageTypeInput
+import edu.ie3.datamodel.models.voltagelevels.GermanVoltageLevelUtils
+import edu.ie3.simona.ontology.messages.flex.MinMaxFlexibilityMessage.ProvideMinMaxFlexOptions
+import edu.ie3.util.TimeUtil
+import edu.ie3.util.scala.quantities.Sq
+import scala.Option
+import spock.lang.Shared
+import spock.lang.Specification
+import squants.energy.*
+
+import static edu.ie3.util.quantities.PowerSystemUnits.*
+import static tech.units.indriya.quantity.Quantities.getQuantity
+
+class StorageModelTest extends Specification {
+
+  @Shared
+  StorageInput inputModel
+  @Shared
+  static final Double TOLERANCE = 1e-10
+
+  def setupSpec() {
+    def nodeInput = new NodeInput(
+    UUID.fromString("ad39d0b9-5ad6-4588-8d92-74c7d7de9ace"),
+    "NodeInput",
+    OperatorInput.NO_OPERATOR_ASSIGNED,
+    OperationTime.notLimited(),
+    getQuantity(1d, PU),
+    false,
+    NodeInput.DEFAULT_GEO_POSITION,
+    GermanVoltageLevelUtils.LV,
+    -1)
+
+    def typeInput = new StorageTypeInput(
+    UUID.fromString("fbee4995-24dd-45e4-9c85-7d986fe99ff3"),
+    "Test_StorageTypeInput",
+    getQuantity(100d, EURO),
+    getQuantity(101d, EURO_PER_MEGAWATTHOUR),
+    getQuantity(100d, KILOWATTHOUR),
+    getQuantity(13d, KILOVOLTAMPERE),
+    0.997,
+    getQuantity(10d, KILOWATT),
+    getQuantity(0.03, PU_PER_HOUR),
+    getQuantity(0.9, PU),
+    )
```

Comment:
see testdata discussion

Reply:
By the way: Shouldn't we use the testdata here?

Reply:
It's currently not possible to use the scala test data here in groovy. The groovy test should be converted to a scalatest anyways at some point, but for now it's more urgent to get the StorageAgent into dev, imho

Reply:
Alright, I agree.